test/e2e.sh

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env bash
+
+set -Eeuxo pipefail
+
+readonly REPO_ROOT="${REPO_ROOT:-$(git rev-parse --show-toplevel)}"
+
+run_ct_container() {
+    echo 'Running ct container...'
+    docker run --rm --interactive --detach --network host --name test-cont \
+        --volume "$(pwd):/go/src/github.com/mattermost/mattermost-operator" \
+        --workdir "/go/src/github.com/mattermost/mattermost-operator" \
+        "golang:1.12.2" \
+        cat
+    echo
+}
+
+docker_exec() {
+    docker exec --interactive test-cont "$@"
+}
+
+
+run_kind() {
+    echo "Download kind binary..."
+    curl -sSLo kind https://github.com/kubernetes-sigs/kind/releases/download/"${KIND_VERSION}"/kind-linux-amd64
+    chmod +x kind
+    sudo mv kind /usr/local/bin/kind
+
+    kind --version
+
+    echo "Download kubectl..."
+    curl -sSLo kubectl https://storage.googleapis.com/kubernetes-release/release/"${K8S_VERSION}"/bin/linux/amd64/kubectl
+    chmod +x kubectl
+    sudo cp kubectl /usr/local/bin/
+    docker cp kubectl test-cont:/usr/local/bin/
+    echo
+
+    echo "Create Kubernetes cluster with kind..."
+    kind create cluster --config test/kind-config.yaml --wait 5m
+
+    echo "Export kubeconfig..."
+    # shellcheck disable=SC2155
+    export KUBECONFIG="$(kind get kubeconfig-path)"
+    cp "$(kind get kubeconfig-path)" ~/.kube/config
+    echo
+
+    echo 'Copying kubeconfig to container...'
+    local kubeconfig
+    kubeconfig="$(kind get kubeconfig-path)"
+    docker_exec mkdir /root/.kube
+    docker cp "$kubeconfig" test-cont:/root/.kube/config
+    docker_exec kubectl cluster-info
+    echo
+
+    echo -n 'Waiting for cluster to be ready...'
+    until ! grep --quiet 'NotReady' <(kubectl get nodes --no-headers); do
```

Comment:
nit: could this ever hang permanently?

Reply:
should not, this is just a safe check because at this point the cluster should be ready, because I've set a waiting time in the cluster creation